NOTES-CMAKE.md

```diff
@@ -0,0 +1,90 @@
+Notes for CMake generator
+===========================
+
+Requirement details
+-------------------
+
+In addition to the obvious CMake requirement, these are required as well:
+
+## Perl
+
+We recommend Strawberry Perl, or alternatively ActiveState Perl on Windows.
+Use the appropriate package manager for Unix systems to acquire Perl.
+
+please see the [Notes for Windows platforms](NOTES-WINDOWS.md) or [Notes for Unix platforms](NOTES-UNIX.md).
+
+
+## Windows
+
+### Netwide Assembler (NASM) on Windows
+
+NASM is the only currently supported assembler on Windows. It is available from <https://www.nasm.us> and has great support with CMake.
+
+Quick start
+-----------
+
+ 1. Install Perl (and NASM is necessary)
+
+ 3. Make sure CMake, Perl, and NASM are all on your PATH.
```

Comment:
Please don't put anything in the PATH.
CMake has [find_program](https://cmake.org/cmake/help/latest/command/find_program.html) function for this purpose or [find_package](https://cmake.org/cmake/help/latest/command/find_package.html).
If OpenSSL really needs to have those in the system PATH, we can discover where the binaries are and
add them to PATH on our site using[ cmake -E env](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-E-arg-env) .
There are so many projects in the world still editing the PATH that my PATH becomes my workspace recently (whatever project I clone I need to change the PATH)...

Reply:
Uhmmmm, on Unix family platforms, it's *normal* to have installed programs in the PATH, and to adjust your PATH when you have programs in non-default locations.  On those platforms, it's annoying to have to deal with other environment to fulfill the same purpose, or even worse, to have to specify root environment variables for each and every program.

I understand that on Windows, because there's no sensible default location where all programs are located, constantly adjusting the PATH is quite annoying, and it's true that CMake has other mechanisms (such as the fairly newly added registry search).

This line could probably be expressed in terms of having *installed* those programs and ensured that they are available in whatever form that's feasible for the target platform